davitpy/pydarn/plotting/rti.py

```diff
@@ -711,9 +713,22 @@ def plot_cpid(ax, times, cpid, mode):
     ax.plot_date(date2num(times), np.arange(len(times)),
                  fmt='w', tz=None, xdate=True, ydate=False, alpha=0.0)
 
+    # Initialize CPID change counter
+    cpid_change = 0
     # Label the CPIDs.
     for i in range(0, len(times)):
         if(cpid[i] != oldCpid):
+            cpid_change += 1
+            # If the cpid is changing too much, it won't be readible
+            if (cpid_change > 20):
+                # Clear the current axis
+                ax.cla()
+                # Kick out error messages
+                ax.text(times[i], .5, 'CPID change error; see logging', ha='left', va='center', size=10)
```

Comment:
On this line it should probably be something like `ax.text(times[0], .5, ...` instead. If I set the `cpidchange_lims=1000` or something else, I can effectively move the position of the text that is being plotted.